src/routeProposal/filtering.ts

```diff
@@ -33,15 +35,20 @@ The main purpose of this function is to:
 As we're looping all here, it also does a number of other things to avoid unnecessary loops later:
 - parsePoolPairData for Direct pools
 - store token decimals for future use
+- We also create and return a pools dictionary that includes all pools which is useful for future pool filtering
 */
 export function filterPoolsOfInterest(
     allPools: SubgraphPoolBase[],
     tokenIn: string,
     tokenOut: string,
     maxPools: number,
+    chainId: number,
     currentBlockTimestamp = 0
-): [PoolDictionary, string[]] {
-    const poolsDictionary: PoolDictionary = {};
+): [PoolDictionary, string[], PoolDictionary] {
+    // This will include all pools
+    const poolsAllDictionary: PoolDictionary = {};
```

Comment:
Personally I'd create this dictionary outside of this function and then pass it in to be filtered. We're overloading what the purpose of this function is by doing this.


My solution would look something like this:
```
const poolsDict = Object.fromEntries(pools
  .map((pool) => [pool.id, parseNewPool(pool, swapOptions.timestamp)])
  .filter(([,pool])=> pool !== undefined)
)
const [poolsFilteredDict, hopTokens] =
    filterPoolsOfInterest(
        poolsDict,
        tokenIn,
        tokenOut,
        swapOptions.maxPools,
    );
```

Reply:
Created a new function `parseToPoolsDict`. This also filters out pools with 0 balance.